docker/Dockerfile.ubuntu-glibc

```diff
@@ -63,6 +63,17 @@ RUN if [ "${DISTRO}" = "xenial" ]; then \
     apt install -y cmake \
     ; fi
 
+# Install C++17 compatible compiler if running on xenial
+RUN if [ "${DISTRO}" = "xenial" ]; then \
+    apt-get install -y software-properties-common \
+    && add-apt-repository ppa:ubuntu-toolchain-r/test \
+    && apt update \
+    && apt install g++-7 -y \
+    && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 60 \
+                           --slave /usr/bin/g++ g++ /usr/bin/g++-7 \
+    && update-alternatives --config gcc \
```

Comment:
Don't you need:

```
       --set name path
              Set the program path as alternative for name.  This is equivalent to --config but is non-interactive and thus scriptable.
```

Instead of:


```
       --config name
              Show available alternatives for a link group and allow the user to interactively select which one to use. The link group is updated.
```

Reply:
Ah, seems this isn't even necessary. When I tried it in a vm:
```
$ update-alternatives --config gcc
There is only one alternative in link group gcc (providing /usr/bin/gcc): /usr/bin/gcc-7
Nothing to configure.
```